ComputeExamples.md

```diff
@@ -0,0 +1,676 @@
+10. [Consumer starts a compute job using a free C2D environment](#-consumer-starts-a-compute-job-using-a-free-c2D-environment)
+11. [Check compute status and get download compute results url](#-check-compute-status-and-get-download-compute-results-url)
+
+Let's go through each step.
+
+## 0. Prerequisites
+Before we start it is important that you have all of the necessary prerequisites installed on your computer.
+- **A Unix based operating system (Linux or Mac)**. If you are a Windows user you can try to run linux inside a virtual machine but this is outside of the scope of this article.
+- **Git**. Instructions for installing Git can be found here: https://git-scm.com/book/en/v2/Getting-Started-Installing-Git
+- **Node.js** can be downloaded from here: https://nodejs.org/en/download/
+- **Docker** can be installed from here: https://docs.docker.com/get-docker/. Please note that Docker must run as a non-root user, you can set this up by following these instructions: https://docs.docker.com/engine/install/linux-postinstall/
+
+## 1. Initialize services
+
+Ocean.js uses off-chain services for metadata (Aquarius) and consuming datasets (Provider).
+
+We start by initializing the services. To do this, we clone the Barge repository and run it. This will run the current default versions of [Aquarius](https://github.com/oceanprotocol/aquarius), [Provider](https://github.com/oceanprotocol/provider), and [Ganache](https://github.com/trufflesuite/ganache) with [our contracts](https://github.com/oceanprotocol/contracts) deployed to it.
+
+```bash
+git clone https://github.com/oceanprotocol/barge.git
+cd barge/
+./start_ocean.sh --with-provider2 --no-dashboard --with-c2d
+```
+
+## 2. Create a new node.js project with Typescript
+
+Start by creating a new Node.js project. Open a new terminal and enter the following commands:
+
+```bash
+mkdir compute-quickstart
+cd compute-quickstart
+npm init
+# Answer the questions in the command line prompt
+touch compute.ts
+# On linux press CTRL + D to save
+```
+
+Next, we need to setup our typescript compiler options. Create a new file called `tsconfig.json` in the root of the `compute-quickstart` directory. copy the following content into the file:
+
+```json
+{
+  "compilerOptions": {
+    "resolveJsonModule": true,
+    "moduleResolution": "node",
+    "esModuleInterop": true,
+    "allowSyntheticDefaultImports": true,
+    "lib": [
+      "ESNext",
+      "dom"
+    ],
+    "declaration": true,
+    "module": "ESNext",
+    "target": "ESNext",
+    "removeComments": false,
+    "experimentalDecorators": true,
+    "preserveConstEnums": true,
+    "outDir": "./",
+    "sourceMap": true,
+    "declarationDir": "./d"
+  },
+  "include": [
+    "compute.ts"
+  ]
+}
+```
+
+Now you can compile your typescript with the following command:
+
+```bash
+tsc
+```
+
+To run your script as we go along, you can use the following command:
+
+```bash
+node compute.js
```

Comment:
nope, won't work. First of all we are creating a `compute.ts` file. And `node` simply can't run those. 

So either we have to tell to run the compiler after each of the following steps, and then run the compiled .js file, or we use [`ts-node`](https://www.npmjs.com/package/ts-node). Would opt for simplicity and tell users to always compile and run.